transport-classes-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -279,6 +281,15 @@ public int registeredChannels() {
         return channels.size();
     }
 
+    @Override
+    public Iterator<Channel> registeredChannelsIterator() {
+        IntObjectMap<AbstractEpollChannel> ch = channels;
+        if (ch.isEmpty()) {
+            return ChannelsReadOnlyIterator.empty();
+        }
+        return new ChannelsReadOnlyIterator<AbstractEpollChannel>(ch.values());
```

Comment:
I think this is not thread-safe as calling values() is not thread-safe

Reply:
It is intended to be non-thread-safe, and be called from within event loop only (same as its `int registeredChannels()` sibling). Do you have use-case in mind? IMO thread-safe channels iterator would make sense for `EventLoopGroup` and not individual event loops.

On the other hand, in tests iterator is incorrectly used outside of eventloop so that needs to be fixed.

Reply:
@mostroverkhov in this case we should check `inEventLoop()` and if not throw.